eq-author-api/schema/resolvers/base.js

```diff
@@ -246,6 +248,39 @@ const Resolvers = {
     },
     createHistoryNote: (root, { input }, ctx) =>
       addEventToHistory(input.id, noteCreationEvent(ctx, input.bodyText)),
+
+    updateHistoryNote: async (root, { input }, ctx) => {
+      const user = ctx.user;
+      const historyItems = await getHistoryById(input.questionnaireId);
+
+      const noteToUpdate = historyItems.find(item => item.id === input.id);
+
+      if (!user.admin && user.id !== noteToUpdate.userId) {
+        throw new Error("User doesnt have access");
+      }
```

Comment:
I think this should be an OR condition; otherwise admins can't update other peoples notes as it'll evaluate to false when they're trying to do so on other peoples.

Reply:
It needs to be `&&` here. If we use and it basically means, `if you are not admin OR you are not the note owner THEN throw the error`. But we want it to throw only if both conditions are true.